custom_components/unifiprotect/switch.py

```diff
@@ -71,50 +78,59 @@ async def async_setup_platform(hass, config, async_add_entities, _discovery_info
 
     switches = []
     for switch_type in config.get(CONF_MONITORED_CONDITIONS):
-        for camera in data.devices:
-            switches.append(UnifiProtectSwitch(data, camera, switch_type, ir_on, ir_off))
+        for camera in coordinator.data:
+            switches.append(
+                UnifiProtectSwitch(coordinator, upv, camera, switch_type, ir_on, ir_off)
+            )
 
     async_add_entities(switches, True)
 
+
 class UnifiProtectSwitch(SwitchDevice):
     """A Unifi Protect Switch."""
 
-    def __init__(self, data, camera, switch_type, ir_on, ir_off):
+    def __init__(self, coordinator, upv, camera, switch_type, ir_on, ir_off):
         """Initialize an Unifi Protect Switch."""
-        self.data = data
+        self.coordinator = coordinator
+        self.upv = upv
         self._camera_id = camera
-        self._camera = self.data.devices[camera]
-        self._name = "{0} {1} {2}".format(DOMAIN.capitalize(), SWITCH_TYPES[switch_type][0], self._camera["name"])
+        self._camera = self.coordinator.data[camera]
+        self._name = "{0} {1} {2}".format(
+            DOMAIN.capitalize(), SWITCH_TYPES[switch_type][0], self._camera["name"]
+        )
         self._unique_id = self._name.lower().replace(" ", "_")
         self._icon = "mdi:{}".format(SWITCH_TYPES.get(switch_type)[1])
         self._ir_on_cmd = ir_on
         self._ir_off_cmd = ir_off
-        self._state = STATE_OFF
         self._camera_type = self._camera["type"]
         self._attr = SWITCH_TYPES.get(switch_type)[2]
         self._switch_type = SWITCH_TYPES.get(switch_type)[2]
         _LOGGER.debug("UnifiProtectSwitch: %s created", self._name)
-        _LOGGER.debug("UnifiProtectSwitch: IR_ON %s IR_OFF %s", self._ir_on_cmd, self._ir_off_cmd)
+        _LOGGER.debug(
+            "UnifiProtectSwitch: IR_ON %s IR_OFF %s", self._ir_on_cmd, self._ir_off_cmd
+        )
 
     @property
     def should_poll(self):
         """Poll for status regularly."""
```

Comment:
When using a data coordinator this should be false, as this entity isn't responsible for updating the data.